scripts/ci/check-connect-data.ts

```diff
@@ -0,0 +1,116 @@
+            const authenticityDevData = await fetchJSON(authenticityDevUrl);
+
+            deviceAuthenticityConfig[deviceKey] = {
+                rootPubKeys: authenticityData.root_pubkeys,
+                caPubKeys: authenticityData.ca_pubkeys,
+                debug: {
+                    rootPubKeys: authenticityDevData.root_pubkeys,
+                    caPubKeys: authenticityDevData.ca_pubkeys,
+                },
+            };
+        }
+
+        const updatedConfigString = `
+            /** THIS FILE IS AUTOMATICALLY UPDATED by script ci/scripts/check-connect-data.ts  */
+            import { DeviceAuthenticityConfig } from './deviceAuthenticityConfigTypes';
+
+            /**
+             * How to update this config or check Sentry "Device authenticity invalid!" error? Please read this internal description:
+             * https://www.notion.so/satoshilabs/Device-authenticity-check-b8656a0fe3ab4a0d84c61534a73de462?pvs=4
+             */
+            export const deviceAuthenticityConfig: DeviceAuthenticityConfig = ${JSON.stringify(deviceAuthenticityConfig, null, 4)};
+          `;
+
+        await fs.writeFile(CONFIG_FILE_PATH, updatedConfigString);
+
+        await exec('yarn', ['prettier', '--write', CONFIG_FILE_PATH]);
+
+        console.log('Configuration updated successfully.');
+
+        console.log('Checking if there were changes.');
+        const changes = await exec('git', ['diff', CONFIG_FILE_PATH]);
+        if (changes.stdout !== '') {
+            console.log('There were changes in keys.');
+            // There were changes in CONFIG_FILE_PATH
+            const hashChanges = await exec('git', [
+                'log',
+                '-1',
+                '--pretty=format:"%H"',
+                '--',
+                CONFIG_FILE_PATH,
+            ]);
+            // We use the hash of the changes to create branch name to avoid use a branch that already exists.
+            const branchName = `chore/update-device-authenticity-config-${hashChanges.stdout.replace(/"/g, '')}`;
+            const commitMessage = 'chore(connect): update device authenticity config';
+            await exec('git', ['checkout', '-b', branchName]);
+            commit({
+                path: ROOT,
+                message: commitMessage,
+            });
+            await exec('git', ['push', 'origin', branchName]);
+
+            await exec('gh', [
+                'pr',
+                'create',
+                '--repo',
+                'trezor/trezor-suite',
+                '--title',
+                `${commitMessage}`,
+                '--body-file',
+                'docs/packages/connect/check-connect-data.md',
+                '--base',
+                'develop',
+                '--head',
+                branchName,
+            ]);
+        }
+    } catch (error) {
+        console.error(`Error updating configuration: ${error.message}`);
```

Comment:
just thinking how useful is to try catch this? also not sure what happens in the CI if we don't do it. This appears to me that if the script fails for any reason, we might not be seeing stac traces in CI log and without catching the error we might be able to see them? Just a nitpick and I am really not sure what is the advisable solution here, would need to run some experiments which might not be worth of doing :D

Reply:
If that fails it will log in the CI the error message and it will give us a clue about what happened.